### PR TITLE
Ryanyae 235 edit timeline fix frontend

### PR DESCRIPTION
**Pull request by ryanyae**
## Description

Fixed the frontend for timeline edit/add bugs
in tandem with https://github.com/UBC-VCL/VCL-content-platform-server/pull/45

## Updates

[Please enter a brief summary of updates to this PR during code review - if any]

## Checklist

- [ ] Ran `npm run lint:fix` to format code
- [ ] Requested at least one reviewer
- [ ] Connected PR to Trello ticket (steps below)
- [ ] Addressed code review comments
- [ ] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

### CODE REVIEW

**Comment by ElisaHu1** [comment]
the ui looks wired lol, what are the save button for? 
image

**Review by ElisaHu1 — APPROVED** [review]

**Comment by ElisaHu1** [comment]
need to work on ui ticket later

**Review by ConnorWon — CHANGES_REQUESTED** [review]

File: src/pages/Timeline/EditTimelineEntry/EditTimelineEntry.tsx [hunk 1]
```diff
@@ -47,59 +55,87 @@ const EditTimelineEntry = () => {
...
+
+      await axios({
+        method: 'put',
+        url: `/api/snapshots/${timeline_id}`,
+        data: {
+          title: timeline.title,
+          description: timeline.description,
+          date: timeline.date,
+          project: timeline.project,
+          author: currUser.username.replace(/['"]+/g, ''),
```
Comment (ConnorWon) [inline src/pages/Timeline/EditTimelineEntry/EditTimelineEntry.tsx:94]: EditTimeline page isn't working anymore and I think it may be because of this line

Comment (ryanyae) [reply src/pages/Timeline/EditTimelineEntry/EditTimelineEntry.tsx:94]: Oh sorry, I wanted to spend a little longer with the code before requesting for reviews again. I'll let you know when I am done

Comment (ConnorWon) [reply src/pages/Timeline/EditTimelineEntry/EditTimelineEntry.tsx:94]: Oh shoot that's mb

**Review by ConnorWon — COMMENTED** [review]
Looking good just one thing.
1. When I try to edit a timeline entry, the Description textarea isn't pre-populated. Also, it doesn't seem to get updated after saving my changes

https://github.com/UBC-VCL/VCL-content-platform/assets/120760211/f5629afa-ef59-407a-a9c3-789fe6e866e9

**Comment by ConnorWon** [comment]
There still seems to be a small problem with the description in EditTimelineEntry. If you create a description that includes '\n':
1.  When you open EditTimeline for that entry the description is displayed as a comma-separated list
2. Also, you can't seem to update the entry if you don't change the description


https://github.com/UBC-VCL/VCL-content-platform/assets/120760211/fb8e93c5-ce78-4a45-8f67-74ad99ce8fcc

**Comment by ConnorWon** [comment]
Sorry to keep nitpicking, but the links for the timeline entry now have an issue when it doesn't start with "https://".
I believe you can fix that if you add a conditional similar to this:
Screenshot 2024-05-01 at 11 09 24 PM

**Comment by ryanyae** [comment]
> Sorry to keep nitpicking, but the links for the timeline entry now have an issue when it doesn't start with "https://". I believe you can fix that if you add a conditional similar to this: Screenshot 2024-05-01 at 11 09 24 PM

Yea I was thinking that either the links be hard typed with https:// or we handle that, but I will change the implementation for us to handle it.

**Review by ConnorWon — APPROVED** [review]

**Pull request merged**